src/server.py

```diff
@@ -0,0 +1,38 @@
+"""Implementation of server."""
+
+import socket
+
+
+def server():
+    """Create a server."""
+    address = ('127.0.0.1', 5000)
+    server = socket.socket(socket.AF_INET,
+                           socket.SOCK_STREAM,
+                           socket.IPPROTO_TCP)
+    server.bind(address)
+    server.listen(1)
+    try:
+        while True:
+            conn, addr = server.accept()
+            message = message_handle(conn)
+            conn.sendall(message.encode('utf8'))
+
+    except (KeyboardInterrupt, BrokenPipeError):
+        print ('Shutting Down')
+        server.close()
+
+
+def message_handle(message):
+    """Handle input message relative to buffer length."""
+    buffer_length = 8
+    message_complete = False
+    output = ''
+    while not message_complete:
+        part = message.recv(buffer_length)
+        output += part.decode('utf8')
```

Comment:
Couple of improvements to be made here.

First, in terms of efficiency, it is much better in Python to append chunks of strings to a list, then use the str.join method to join them all at the end, rather than concatenate one at a time like this.

Secondly, this will break sometimes on non-unicode characters. So you're decoding each chunk as you get it. Since unicode characters can take up multiple bytes, what happens when a multiple-byte unicode character isn't received all the way in one chunk? i.e. the buffer length cuts it in half?

Rather than decoding one chunk at a time, what would be a better approach?